src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -160,25 +147,32 @@ public JedisSentinelPool(String masterName, Set<String> sentinels,
       final String user, final String password, final int database, final String clientName,
       final int sentinelConnectionTimeout, final int sentinelSoTimeout, final String sentinelUser,
       final String sentinelPassword, final String sentinelClientName) {
+    this(masterName, parseHostAndPorts(sentinels), poolConfig,
+        DefaultJedisClientConfig.builder().withConnectionTimeoutMillis(connectionTimeout)
+            .withSoTimeoutMillis(soTimeout).withInfiniteSoTimeoutMillis(infiniteSoTimeout).withUser(user)
+            .withPassword(password).withDatabse(database).withClientName(clientName).build(),
+        DefaultJedisClientConfig.builder().withConnectionTimeoutMillis(sentinelConnectionTimeout)
+            .withSoTimeoutMillis(sentinelSoTimeout).withUser(sentinelUser).withPassword(sentinelPassword)
+            .withClientName(sentinelClientName).build()
+    );
+  }
+
+  public JedisSentinelPool(String masterName, Set<HostAndPort> sentinels,
+      final GenericObjectPoolConfig<Jedis> poolConfig, final JedisClientConfig masteClientConfig,
+      final JedisClientConfig sentinelClientConfig) {
 
     this.poolConfig = poolConfig;
-    this.connectionTimeout = connectionTimeout;
-    this.soTimeout = soTimeout;
-    this.infiniteSoTimeout = infiniteSoTimeout;
-    this.user = user;
-    this.password = password;
-    this.database = database;
-    this.clientName = clientName;
-    this.sentinelConnectionTimeout = sentinelConnectionTimeout;
-    this.sentinelSoTimeout = sentinelSoTimeout;
-    this.sentinelUser = sentinelUser;
-    this.sentinelPassword = sentinelPassword;
-    this.sentinelClientName = sentinelClientName;
+    this.masterClientConfig = masteClientConfig;
+    this.sentinelClientConfig = sentinelClientConfig;
 
     HostAndPort master = initSentinels(sentinels, masterName);
     initPool(master);
   }
 
+  private static Set<HostAndPort> parseHostAndPorts(Set<String> strings) {
+    return strings.parallelStream().map(str -> HostAndPort.parseString(str)).collect(Collectors.toSet());
```

Comment:
I would avoid `parallelStream` here:
1. I don't really think it's that intensive of an operation that you are going to get a gain
2. It's going to use the shared forkjoin pool and might affect other operations

Also we can use the method reference for the map operation.

```suggestion
    return strings.stream().map(HostAndPort::parseString).collect(Collectors.toSet());
```